File: backend/Models/ContainerStorageFee.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Backend.Models
{
    /// <summary>
    /// Tracks container storage fees and duration at ports
    /// </summary>
    public class ContainerStorageFee
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [StringLength(50)]
        public string ContainerId { get; set; }

        [Required]
        public int PortId { get; set; }

        [Required]
        public DateTime StorageStartDate { get; set; }

        public DateTime? StorageEndDate { get; set; }

        [Required]
        [Column(TypeName = "decimal(10,2)")]
        public decimal DailyStorageRate { get; set; }

        public int TotalDays { get; set; } = 0;

        [Column(TypeName = "decimal(15,2)")]
        public decimal TotalFees { get; set; } = 0;

        [Required]
        [StringLength(20)]
        public string FeeStatus { get; set; } = "Calculating"; // Calculating, Finalized, Paid

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        [ForeignKey("ContainerId")]
        public virtual Container Container { get; set; }

        [ForeignKey("PortId")]
        public virtual Port Port { get; set; }
    }
}

File: frontend/src/types/containerStorageFee.ts
/**
 * ContainerStorageFee type definitions aligned with backend DTOs
 */

export interface ContainerStorageFee {
  id: number
  containerId: string
  portId: number
  storageStartDate: string
  storageEndDate: string | null
  dailyStorageRate: number
  totalDays: number
  totalFees: number
  feeStatus: string
}

export interface ContainerStorageFeeCreateUpdate {
  containerId: string
  portId: number
  storageStartDate: string
  storageEndDate: string | null
  dailyStorageRate: number
  feeStatus: string
}

export type FeeStatus = 'Pending' | 'Paid' | 'Overdue' | 'Waived'
